### PR TITLE
Service Endpoints

### PR DESCRIPTION
**Pull request by robgil-dds**
This adds the initial configuration necessary for service endpoints. Service endpoints allow a service to be exposed only within specific subnets or be ip restricted. The initial module to support this is the storage module.

### CODE REVIEW

**Comment by robgil-dds** [comment]
Still WIP. Ran in to an issue with Redis refusing to share a network.

```
Error: Error issuing create request for Redis Cache cloudzero-jedidev-redis (resource group cloudzero-jedidev-redis): redis.Client#Create: Failure sending request: StatusCode=400 -- Original Error: Code="BadRequest" Message="The subnet must be empty except for redis caches. This subnet cannot be used because it contains '/subscriptions/65453559-4bfd-413a-8593-6c9bd09bb47a/resourceGroups/MC_cloudzero-jedidev-vpc_cloudzero-jedidev-k8s_eastus/providers/Microsoft.Compute/virtualMachineScaleSets/aks-default-54410534-vmss/virtualMachines/0/networkInterfaces/aks-default-54410534-vmss/ipConfigurations/ipconfig1'.\r\nRequestID=5a811c9c-00e7-4278-988e-1a74097324cd"
```

**Comment by robgil-dds** [comment]
Redis is all set now except for the subnet configuration. There doesn't seem to be a "Microsoft.Redis" service endpoint name. I'm not sure which service endpoint redis comes under, or if this subnet configuration is even necessary. Perhaps it just runs happily in its own dedicated subnet.

File: terraform/providers/dev/variables.tf [hunk 1]
```diff
@@ -79,3 +90,24 @@ variable "admin_users" {
...
+  }
+}
+
+variable "storage_admin_whitelist" {
+  type = map
+  default = {
+    "Rob Gil"           = "66.220.238.246"
+    "Dan Corrigan Work" = "108.16.207.173"
+  }
+}
```
Comment (dandds) [inline terraform/providers/dev/variables.tf]: Do we want to merge with these or wait until we can split them out in a private repo?

File: terraform/providers/dev/k8s-test.tf-old [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    ignore_changes = [
+      default_node_pool.0.node_count
+    ]
+  }
+
+  tags = {
+    environment = var.environment
+    owner       = var.owner
+  }
+}
```
Comment (dandds) [inline terraform/providers/dev/k8s-test.tf-old]: Did you mean to commit this?

File: terraform/providers/dev/variables.tf (hunk 1 above)
Comment (robgil-dds) [reply terraform/providers/dev/variables.tf]: I'd prefer getting this all in to the private repo.

File: terraform/providers/dev/k8s-test.tf-old (hunk 2 above)
Comment (robgil-dds) [reply terraform/providers/dev/k8s-test.tf-old]: Nope! Removing.

**Pull request merged**